src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py

```diff
@@ -655,6 +655,8 @@ def create_service_principal_for_rbac(
                     raise
 
     if expanded_view:
+        logger.warning("'--expanded-view' is deprecating and will be removed in a future release. "
+                       "You can get the same information using 'az cloud show'")
```

Comment:
Consider making it help=argparse.SUPPRESS so that it no longer shows with -h (hidden but continuing to function, unlike if you made it ignore_type).  That way no one coming to the CLI now would use it.